common/src/main/cpp/extensions/openxr_fb_hand_tracking_mesh_extension_wrapper.cpp

```diff
@@ -0,0 +1,309 @@
+				}
+			}
+		}
+	}
+}
+
+void OpenXRFbHandTrackingMeshExtensionWrapper::set_use_scale_override(Hand p_hand, bool p_use) {
+	hand_tracking_scale[p_hand].overrideHandScale = p_use;
+}
+
+void OpenXRFbHandTrackingMeshExtensionWrapper::set_scale_override(Hand p_hand, float p_scale) {
+	hand_tracking_scale[p_hand].overrideValueInput = p_scale;
+}
+
+bool OpenXRFbHandTrackingMeshExtensionWrapper::fetch_hand_mesh_data(Hand p_hand) {
+	ERR_FAIL_COND_V_MSG(!is_enabled(), false, "OpenXR extension XR_FB_hand_tracking_mesh is not available");
+
+	XrHandTrackingMeshFB hand_mesh;
+	hand_mesh.type = XR_TYPE_HAND_TRACKING_MESH_FB;
+	hand_mesh.next = nullptr;
+	hand_mesh.jointCapacityInput = 0;
+	hand_mesh.vertexCapacityInput = 0;
+	hand_mesh.indexCapacityInput = 0;
+	XrHandTrackerEXT hand_tracker = reinterpret_cast<XrHandTrackerEXT>(get_openxr_api()->get_hand_tracker(p_hand));
+
+	XrResult result = xrGetHandMeshFB(hand_tracker, &hand_mesh);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("Failed to retreive capacity inputs for OpenXR XR_FB_hand_tracking_mesh extension, error code: ", result);
+		return false;
+	}
+
+	hand_mesh.jointCapacityInput = hand_mesh.jointCountOutput;
+	hand_mesh.vertexCapacityInput = hand_mesh.vertexCountOutput;
+	hand_mesh.indexCapacityInput = hand_mesh.indexCountOutput;
+
+	hand_mesh.jointBindPoses = reinterpret_cast<XrPosef *>(memalloc(sizeof(XrPosef) * hand_mesh.jointCapacityInput));
+	hand_mesh.jointRadii = reinterpret_cast<float *>(memalloc(sizeof(float) * hand_mesh.jointCapacityInput));
+	hand_mesh.jointParents = reinterpret_cast<XrHandJointEXT *>(memalloc(sizeof(XrHandJointEXT) * hand_mesh.jointCapacityInput));
+	hand_mesh.vertexPositions = reinterpret_cast<XrVector3f *>(memalloc(sizeof(XrVector3f) * hand_mesh.vertexCapacityInput));
+	hand_mesh.vertexNormals = reinterpret_cast<XrVector3f *>(memalloc(sizeof(XrVector3f) * hand_mesh.vertexCapacityInput));
+	hand_mesh.vertexUVs = reinterpret_cast<XrVector2f *>(memalloc(sizeof(XrVector2f) * hand_mesh.vertexCapacityInput));
+	hand_mesh.vertexBlendIndices = reinterpret_cast<XrVector4sFB *>(memalloc(sizeof(XrVector4sFB) * hand_mesh.vertexCapacityInput));
+	hand_mesh.vertexBlendWeights = reinterpret_cast<XrVector4f *>(memalloc(sizeof(XrVector4f) * hand_mesh.vertexCapacityInput));
+	hand_mesh.indices = reinterpret_cast<int16_t *>(memalloc(sizeof(int16_t) * hand_mesh.indexCapacityInput));
+
+	result = xrGetHandMeshFB(hand_tracker, &hand_mesh);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("Failed to retreive hand mesh data for OpenXR XR_FB_hand_tracking_mesh extension, error code: ", result);
+		return false;
+	}
+
+	// convert to clockwise winding order to cull correct face side
+	for (int i = 0; i < hand_mesh.indexCapacityInput; i += 3) {
+		SWAP(hand_mesh.indices[i], hand_mesh.indices[i + 2]);
+	}
+
+	hand_mesh_data[p_hand] = hand_mesh;
+
+	return true;
+}
+
+Ref<ArrayMesh> OpenXRFbHandTrackingMeshExtensionWrapper::get_mesh(Hand p_hand) {
```

Comment:
Seems like this generates a new `Mesh` every time it's called. Since the underlying data is not supposed to change for the lifetime of the `XrInstance`, should this logic cache the generated `Mesh` and return it on subsequent calls?